R/prep_and_calc_loan_book.R

```diff
@@ -103,17 +103,14 @@ run_prep_calculation_loans <- function(credit_type = "outstanding") {
   )
 
   #### Wrangle and prepare data-------------------------------------------------
-
-  # TODO: what to do with negative credit limits?
-  matched_non_negative <- matched %>%
-    dplyr::mutate(
-      loan_size_outstanding = dplyr::if_else(
-        .data$loan_size_outstanding < 0, 0, .data$loan_size_outstanding
-      ),
-      loan_size_credit_limit = dplyr::if_else(
-        .data$loan_size_credit_limit < 0, 0, .data$loan_size_credit_limit
-      )
-    )
+  # ADO 2690 - remove rows with negative loan values (not allowed in P4B)
+  if (credit_type == "outstanding") {
+    matched_non_negative <- matched %>%
+      dplyr::filter(.data$loan_size_outstanding >= 0)
+  } else {
+    matched_non_negative <- matched %>%
+      dplyr::filter(.data$loan_size_credit_limit >= 0)
+  }
```

Comment:
I wonder if this dropping of rows should be related some user facing information (log message or warning), indicating which companies are affected?

Reply:
done